all-repos.yaml

```diff
@@ -206,6 +206,7 @@
 - https://github.com/adamchainz/django-upgrade
 - https://github.com/scop/pre-commit-shfmt
 - https://github.com/scop/pre-commit-perlcritic
+- https://github.com/scop/wrun
```

Comment:
this downloads tools at runtime rather than at build time so it's not a good fit for the framework -- perhaps you can chime in on https://github.com/pre-commit/pre-commit/issues/1453 and https://github.com/pre-commit/pre-commit/pull/3020 which are trying to solve this more generically (and correctly)

Reply:
the problem is more that there's no write protection to the pre-commit cache at runtime (intentionally) and so writes in general break the cache integrity and at worst concurrent writes permanently poison the cache and so writing at runtime is unsupported

Reply:
A workaround for that could be if pre-commit.ci provided a way to cache additional paths, not just the pre-commit cache.  Whatever happens in them would be the responsibility of something else besides pre-commit. Can imagine that this would likely make sharing such additional caches pretty much a no go, as well as their retention suboptimal.

Reply:
it's not a pre-commit.ci issue -- it's a pre-commit issue.  writes to pre-commit's cache at runtime are unsafe

Reply:
> I'll likely remove the helper option to (ab)use a subdir in pre-commit's cache. 

FTR, this option has been [removed since](https://github.com/scop/wrun/commit/dd08280265481545a6563835a6317008597b97b5), wrun no longer helps or recommends the user in any way to touch pre-commit's cache.

So in that sense I suppose the blocker against adding this in pre-commit's supported hooks list is now gone, and the addition could be reconsidered. My branch is still there.